src/Interpreters/Context.cpp

```diff
@@ -2720,7 +2877,7 @@ bool Context::hasDistributedDDL() const
 
 void Context::setDDLWorker(std::unique_ptr<DDLWorker> ddl_worker)
 {
-    auto lock = getLock();
+    auto lock = getLocalLock();
```

Comment:
I think the next line and l. 2884 needs the global lock.

Reply:
Fixed in https://github.com/ClickHouse/ClickHouse/pull/55352.